### DISCUSSION TITLE
Can't save PNG with exact specific palette.

### DISCUSSION THREAD

**Discussion by weegee720** [dlemstra/Magick.NET#718] [Help]
No matter what I do, MagickNET saves PNG file with changed (likely optimized) palette. I've found out that this behavior can be changed with define "preserve-colormap". But after I use this define I get the following exception:

ImageMagick.MagickCoderErrorException: 'Valid palette required for paletted images' 

I've tried settings the palette using image.setColorMap and image.map(colorMap) without any success.

Preserving the palette is an absolute must, and it can't be changed. The project where these files will be used is a legacy project relying in 256-color indexed color bitmaps.

All my code for reference:

        public static void Save(string directory, (int width, int height) size, MagickColor[] colorMap, List<ImageEx> images)
	{
            var outputBitmap = new MagickImage();
            outputBitmap.PreserveColorType();
            outputBitmap.Read("z:\\Git\\SpritePacker\\palette.bmp");
            outputBitmap.Format = MagickFormat.Png8;
            outputBitmap.Resize(size.width, size.height);

            outputBitmap.ColormapSize = 256;
            for (int i = 0; i < 256; i++)
                outputBitmap.SetColormap(i, colorMap[i]);

            foreach (var image in images)
                outputBitmap.Composite(image.Image, new PointD(image.AtlasX, image.AtlasY));

            outputBitmap.Settings.SetDefine(MagickFormat.Png8, "preserve-colormap", "true");

            outputBitmap.Write(directory + "/SpriteAtlas.png");
	}

**Comment by t246246** [comment]
I think this is the issue that is not solved yet: https://github.com/ImageMagick/ImageMagick/issues/416
